File: backend/app/schemas/audit.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class AuditEventResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    occurred_at: datetime
    actor_user_id: UUID | None
    action: str
    entity_type: str
    entity_id: UUID | None
    organization_id: UUID | None
    business_unit_id: UUID | None
    division_id: UUID | None
    department_id: UUID | None
    request_id: UUID | None
    source_ip: str | None
    changes: dict[str, Any] | None
    reason: str | None
    source: str

File: frontend/src/api/audit.ts
import { apiRequest } from './client'

export type AuditEvent = {
  id: string; occurred_at: string; actor_user_id: string | null; action: string;
  entity_type: string; entity_id: string | null; organization_id: string | null;
  request_id: string | null; source_ip: string | null; changes: Record<string, unknown> | null;
  reason: string | null; source: string
}
export type AuditFilters = { entity_type?: string; action?: string; actor_user_id?: string; entity_id?: string; limit?: number; offset?: number }

export const auditApi = {
  list: (filters: AuditFilters = {}) => {
    const query = new URLSearchParams()
    Object.entries(filters).forEach(([key, value]) => { if (value !== undefined && value !== '') query.set(key, String(value)) })
    return apiRequest<AuditEvent[]>(`/audit/events?${query}`)
  },
  get: (id: string) => apiRequest<AuditEvent>(`/audit/events/${id}`),
}
